code/game/machinery/mind_machine.dm

```diff
@@ -0,0 +1,846 @@
+	podsConnected = null
+	currentlySwapping = null
+	lockedPods = FALSE
+	occupantScan = FALSE
+	occupantOne = null
+	occupantTwo = null
+	swapProgress = 0
+	if(connectOne || connectTwo || podsConnected)
+		connectOne.connectedHub = null
+		connectTwo.connectedHub = null
+		connectOne.podNumber = 0
+		connectTwo.podNumber = 0
+		connectOne = null
+		connectTwo = null
+	. = ..()
+
+/obj/machinery/mind_machine/mind_machine_hub/attackby(var/obj/item/A as obj, var/mob/user as mob)
+	..()
+	if(istype(A, /obj/item/bluespace_crystal/flawless))
+		user.drop_item(A, src)
+		bluespaceConduit += 300
+		to_chat(user, "The bluespace conduit flashes violently before calming to a glow.")
+		qdel(A)
+		return
+	if(istype(A, /obj/item/bluespace_crystal/artificial))
+		user.drop_item(A, src)
+		bluespaceConduit += 1
+		to_chat(user, "The crystal is assimilated into the conduit!")
+		qdel(A)
+		return
+	if(istype(A, /obj/item/bluespace_crystal))
+		user.drop_item(A, src)
+		bluespaceConduit += 3
+		to_chat(user, "The crystal is assimilated into the conduit!")
+		qdel(A)
+		return
+	if(istype(A, /obj/item/device/soulstone))
+		if(soulShardSafety != FALSE)
+			to_chat(user, "That slot is full!")
+			return
+		if(A.contents.len)
+			to_chat(user, "The stone must be empty to function properly.") //prevents deleting players in stones
+			return
+		else
+			user.drop_item(A, src)
+			to_chat(user, "The stone is pulled into the machine. You can hear crunching and a quiet sizzle from inside.")
+			qdel(A)
+			soulShardSafety = TRUE
+
+/obj/machinery/mind_machine/mind_machine_hub/attack_hand(user as mob)
+	if(podsConnected == FALSE || connectOne == FALSE || connectTwo == FALSE)
+		podsConnected = FALSE
+		connectOne = FALSE
+		connectTwo = FALSE
+		findConnections() //No UI without pods
+	if(!podsConnected)
+		to_chat(user, "Pod connection error.")
+		return
+	ui_interact(user)
+
+//////UI stuff///////////////
+
+/obj/machinery/mind_machine/mind_machine_hub/ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open = NANOUI_FOCUS)
+	var/list/data = list(
+			"nameOne" = occupantNameOne,
+			"nameTwo" = occupantNameTwo,
+			"statOne" = occupantStatOne,
+			"statTwo" = occupantStatTwo,
+			"mindTypeOne" = mindTypeOne,
+			"mindTypeTwo" = mindTypeTwo,
+			"lockedPods" = lockedPods,
+			"bluespaceConduit" = bluespaceConduit,
+			"soulShardSafety" = soulShardSafety,
+			"manipRating" = manipRating,
+			"swapProgress" = swapProgress,
+			"errorMessage" = errorMessage
+	)
+
+	ui = nanomanager.try_update_ui(user, src, ui_key, ui, data, force_open)
+
+	if(!ui)
+		ui = new(user, src, ui_key, "mind_machine.tmpl", "Mind Machine", 350,500)
+		ui.set_initial_data(data)
+		ui.open()
+	ui.set_auto_update(1)
+
+/obj/machinery/mind_machine/mind_machine_hub/Topic(href, href_list)
+	if(href_list["scan_start"])
+		if(!currentlySwapping)
+			scanOccupants()
+			return TRUE
+		else
+			errorMessage = "Currently active: Please wait"
```

Comment:
>keeping a var just for this

Seems like a recipe to disaster

Reply:
Yeah it makes more sense to make it a local var

Reply:
This will be the hardest thing to fix and I need more info as to what's wrong with it to make sure I do it right. Do you just want me to have it use to_chat instead? I like having it on the UI but I don't think I'm good enough with nano to avoid a bunch of strings.
As for disappearing after 5 seconds, I intentionally left it to linger because if someone uses a delayed swap and gets sent somewhere I want at least a little evidence on screen of what happened.